main/src/ca/uwaterloo/flix/language/phase/jvm/GenFunctionInterfaces.scala

```diff
@@ -52,23 +54,35 @@ object GenFunctionInterfaces {
       return None
     }
 
-    // The function arity is simply the number of type arguments.
-    val arity = args.length
+    // `JvmType` of the continuation interface for `tpe`
+    val continuationType = JvmOps.getContinuationType(tpe)
 
-    // The name of the functional interface is of the form:
-    // Fn1$Int$Bool,
-    // Fn2$Int$Int$Bool,
-    // Fn3$Obj$Obj$Obj$Bool, etc.
-    // TODO: Probably use a helper function to compute this?
-    val name = "Fn" + arity + "$"
+    // `JvmType` of the functional interface for `tpe`
+    val functionType = JvmOps.getFunctionType(tpe)
 
-    for (arg <- args) {
-      // do something with the arguments ...
+    // Class visitor
+    val visitor = new ClassWriter(0)
 
+    // Interfaces to be extended
+    val extendedInterfaced = Array(continuationType.name.toInternalName)
+
+    // Class visitor
+    visitor.visit(JvmOps.JavaVersion, ACC_PUBLIC + ACC_ABSTRACT + ACC_INTERFACE, functionType.name.toInternalName, null,
+      JvmType.Obj.name.toInternalName, extendedInterfaced)
+
+    // Adding setters for each arg
+    args.zipWithIndex.foreach{ case (arg, ind) =>
```

Comment:
Regular for loop, but you are welcome to use zipWithIndex :)

Reply:
See comment above!